### PR TITLE
Use ListAdapter in NotificationModeConfigAdapter.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Use `ListAdapter` in the notification mode adapter.

#### Fixes the following issue(s)

- Fixes #

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/settings/notifications/NotificationModeConfigFragment.kt [hunk 1]
```diff
@@ -15,47 +16,61 @@ import io.reactivex.rxjava3.disposables.Disposable
...
         savedInstanceState: Bundle?,
-    ): View = inflater.inflate(R.layout.fragment_channels_notifications, container, false)
+    ): View {
+        _binding = FragmentChannelsNotificationsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         val recyclerView: RecyclerView = view.findViewById(R.id.recycler_view)
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/settings/notifications/NotificationModeConfigFragment.kt:57]: Shouldn't you be using the `binding` here? `binding` is currently unused.

**Review by Stypox — CHANGES_REQUESTED** [review]
Looks almost good to me, thanks

**Review by Stypox — CHANGES_REQUESTED** [review]
It does not work. Just open "Settings -> Notifications -> Channels" and tick a checkbox to see that everything disappears or the app crashes. Please test your PRs before submitting them (and before submitting any further change), otherwise it's just a waste of time to review things that don't work.

Another thing that you should do is make sure the CI succeeds. In this case it does not succeed because of a misformatted Kotlin file.

**Comment by Isira-Seneviratne** [comment]
@Stypox Sorry about that, fixed.

**Review by Stypox — APPROVED** [review]
Code looks good and works. I pushed a small commit that further simplifies the handling of the disposables. Thank you!

**Pull request merged**